### PR TITLE
[hueemulation] Fix for Alexa failing to discover all devices.

### PR DESCRIPTION
**Pull request by MikeJMajor**
Alexa currently fails to discover devices which have an internal hue id greater than 255. This appears to be due to the value of the uniqueid which is not in the format described in the hue API when the value exceeds 255.

[Reports of issue and fix tested](https://community.openhab.org/t/oh3-hue-emulation-bug/139875)

Signed-off-by: Mike Major <mike_j_major@hotmail.com>

### CODE REVIEW

File: bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/ConfigStore.java [hunk 1]
```diff
@@ -271,15 +271,14 @@ private String getConfiguredHostAddress(InetAddress configuredAddress) {
      */
     private String getHueIDPrefixFromUUID(final String uuid) {
         // Hue API example of a unique id is AA:BB:CC:DD:EE:FF:00:11-XX
-        // XX is generated from the item.
+        // 00:11-XX is generated from the item.
         String prefix = uuid;
         try {
             // Generate prefix if uuid is a randomly generated UUID
             if (UUID.fromString(uuid).version() == 4) {
                 final StringBuilder sb = new StringBuilder(23);
```
Comment (lolodomo) [inline bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/ConfigStore.java:279]: Size is now 17.

**Review by lolodomo — COMMENTED** [review]
Is it a change that will potentially break OH already setup by users ?

**Comment by MikeJMajor** [comment]
@lolodomo Users might have to rescan their devices following the id change.

**Comment by lolodomo** [comment]
> Users might have to rescan their devices following the id change.

You mean in the hue app? 

I am not sure the timing is good while we are now in the few last days before the release GA with expected restricted merges.

**Comment by MikeJMajor** [comment]
> You mean in the hue app?
> 
> I am not sure the timing is good while we are now in the few last days before the release GA with expected restricted changes/merges.

Its whatever is talking to the Hue Emulator, e.g. Alexa, Logitech Harmony, etc that might need a device rescan depending on what they do with that unique id.

I wasn't expecting this to be merged straight away due to the pending 3.4 release.

**Review by lolodomo — APPROVED** [review]

**Comment by lolodomo** [comment]
Please submit an entry in update.lst to explain the breaking change.
Take the following PR as example of what file to update and how to do it:
https://github.com/openhab/openhab-distro/pull/1449/files

**Comment by azntech** [comment]
Is there plans for this fix to be merged in to the 3.4 branch?

**Pull request merged**